core/metric.go

```diff
@@ -45,8 +53,102 @@ type Metric interface {
 	Unit() string
 }
 
+// NamespaceElement defines the method set for the namespace elements
+type NamespaceElement interface {
+	IsDynamic() bool
+	Value() string
+	Name() string
+	Description() string
+	SetValue(value string)
+	SetName(name string)
+	SetDescription(desc string)
+}
+
+// Namespace defines array of namespace elements
 type Namespace []NamespaceElement
 
+// DynamicElement defines the dynamic element of a namespace
+type DynamicElement struct {
+	Value_       string
+	Description_ string
+	Name_        string
+}
```

Comment:
Should we have the functions for DynamicElement next to it's definition?

I also think it probably makes sense to have all the values be unexported since they are all exposed/modifiable via functions.